### ISSUE TITLE
Apply configuration avoidance recommendations for tasks

### ISSUE THREAD

**Issue by MyDogTom** [Karumi/Shot#128]
### Expected behaviour
All tasks should be created lazily to avoid extra overhead during configuration phase.

### Actual behaviour
Tasks are created eagerly using `getTasks.create` method.

### Steps to reproduce
run `gradle help --scan` on any project that uses Shot. Check "performance -> configuration". You will see that tasks like "executeScreenshotTests" are created, while they should be created only if they are executed.

### Version of the library
4.3.0

### More details

Documentation about configuration avoidance: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

Basically it should be enough to register tasks (`tasks.register`) instead of creating them (`tasks.create`) and make sure that all tasks configuration happens inside `configure` block instead of getting instance of the task.

**Comment by pedrovgs** [comment]
You are absolutely right! Thanks for the report and the pull request. I'm going to review it today!
